files/image_config/rsyslog/rsyslog.conf.j2

```diff
@@ -91,25 +90,28 @@ $RepeatedMsgReduction on
 # Remote syslog logging
 #
 
-# The omfwd plug-in provides the core functionality of traditional message forwarding via UDP and plain TCP.
-# It is a built-in module that does not need to be loaded.
+# The omfwd plug-in provides the core functionality of traditional message
+# forwarding via UDP and plain TCP. It is a built-in module that does not need
+# to be loaded.
 
-{% if SYSLOG_SERVER is defined %}
-{% for server, data in SYSLOG_SERVER.items() %}
-{% set params_list = [] %}
-{% if 'source' in data %}
-{% set dummy = params_list.append('address=' + '"' + data.source|string + '"') %}
-{% endif %}
-{% if 'port' in data %}
-{% set dummy = params_list.append('port=' + '"' + data.port|string + '"') %}
-{% endif %}
-{% if 'vrf' in data and data['vrf'] != "default" %}
-{% set dummy = params_list.append('device=' + '"' + data.vrf|string + '"') %}
-{% endif %}
-{% if params_list %}
-*.* action(type="omfwd" target="{{ server }}" protocol="udp" {{ params_list|join(' ') }} template="SONiCFileFormat")
-{% else %}
-*.* action(type="omfwd" target="{{ server }}" protocol="udp" template="SONiCFileFormat")
+{% set servers = SYSLOG_SERVER | d({}) -%}
+{% for server in servers %}
+{% set conf = servers[server] | d({}) -%}
+
+{% set port = conf.get('port', 514) -%}
+{% set proto = conf.get('protocol', 'udp') -%}
+{% set vrf = conf.get('vrf', 'default') -%}
+{% set severity = conf.get('severity', gconf.get('severity', 'notice')) -%}
+{% set filter = conf.get('filter') -%}
+{% set regex = conf.get('filter_regex') -%}
+
+{% set fmodifier = '!' if filter == 'exclude' else '' %}
+{% set device = 'mgmt' if vrf in ['mgmt', 'management'] else 'eth0' %}
```

Comment:
This assumes that syslog server is reachable only through mgmt network or eth0. What if the server is reachable through the data network or OOB mgmt port?

Reply:
oh, yes. I missed the `address` as well. I will fix it.
Good catch!

Reply:
Updated